rust/crd/src/s3logdir.rs

```diff
@@ -0,0 +1,215 @@
+
+        if let Some(endpoint) = self.bucket.endpoint() {
+            result.insert("spark.hadoop.fs.s3a.endpoint".to_string(), endpoint);
+        }
+
+        if let Some(conn) = self.bucket.connection.as_ref() {
+            if let Some(S3AccessStyle::Path) = conn.access_style {
+                result.insert(
+                    "spark.hadoop.fs.s3a.path.style.access".to_string(),
+                    "true".to_string(),
+                );
+            }
+        }
+        result
+    }
+
+    pub fn application_spark_config(&self) -> BTreeMap<String, String> {
+        let mut result = BTreeMap::new();
+        result.insert("spark.eventLog.enabled".to_string(), "true".to_string());
+        result.insert("spark.eventLog.dir".to_string(), self.url());
+
+        let bucket_name = self.bucket.bucket_name.as_ref().unwrap().clone();
+        if let Some(endpoint) = self.bucket.endpoint() {
+            result.insert(
+                format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.endpoint"),
+                endpoint,
+            );
+        }
+
+        if let Some(conn) = self.bucket.connection.as_ref() {
+            if let Some(S3AccessStyle::Path) = conn.access_style {
+                result.insert(
+                    format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.path.style.access"),
+                    "true".to_string(),
+                );
+            }
+
+            if let Some(secret_dir) = self.credentials_mount_path() {
+                // We don't use the credentials at all here but assume they are available
+                result.insert(
+                    format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.access.key"),
+                    format!("$(cat {secret_dir}/{ACCESS_KEY_ID})"),
+                );
+                result.insert(
+                    format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.secret.key"),
+                    format!("$(cat {secret_dir}/{SECRET_ACCESS_KEY})"),
+                );
+                result.insert(
+                    format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.aws.credentials.provider"),
+                    "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider".to_string(),
+                );
+            } else {
+                result.insert(
+                    format!("spark.hadoop.fs.s3a.bucket.{bucket_name}.aws.credentials.provider"),
+                    "org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider".to_string(),
+                );
+            }
+        }
+
+        result
+    }
+
+    fn url(&self) -> String {
+        format!(
+            "s3a://{}/{}",
+            self.bucket.bucket_name.as_ref().unwrap().clone(), // this is guaranteed to exist at this point
+            self.prefix
+        )
+    }
+    pub fn credentials_volume(&self) -> Option<Volume> {
+        self.credentials()
+            .map(|credentials| credentials.to_volume(credentials.secret_class.as_ref()))
```

Comment:
I personally would prefer calling the volume and volume mount something like `<secret_class>-s3-credentials` for better debugging as well as being absolutely sure no paths will clash.
Feel free to ignore this as it's a minor detail

Reply:
Done.